src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/Logging/HTMLCSharpLanguageServerLogHubLoggerProvider.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Composition;
+using System.Diagnostics;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.VisualStudio.LanguageServerClient.Razor.Logging
+{
+    [Shared]
+    [Export(typeof(HTMLCSharpLanguageServerLogHubLoggerProvider))]
+    internal class HTMLCSharpLanguageServerLogHubLoggerProvider : ILoggerProvider
+    {
+        private static readonly string LogFileIdentifier = "HTMLCSharpLanguageServer";
+
+        private LogHubLoggerProvider _loggerProvider;
+
+        private readonly HTMLCSharpLanguageServerLogHubLoggerProviderFactory _loggerFactory;
+
+        // Internal for testing
+        internal HTMLCSharpLanguageServerLogHubLoggerProvider()
+        {
+        }
+
+        [ImportingConstructor]
+        public HTMLCSharpLanguageServerLogHubLoggerProvider(
+            HTMLCSharpLanguageServerLogHubLoggerProviderFactory loggerFactory)
+        {
+            if (loggerFactory == null)
+            {
+                throw new ArgumentNullException(nameof(loggerFactory));
+            }
+
+            _loggerFactory = loggerFactory;
+        }
+
+        public async Task InitializeLoggerAsync()
+        {
+            if (_loggerProvider is null)
+            {
+                _loggerProvider = (LogHubLoggerProvider)await _loggerFactory.GetOrCreateAsync(LogFileIdentifier).ConfigureAwait(false);
```

Comment:
On restart will the logger factory automatically "increment"? Asking because we don't want a server restart to overwrite all the content we previously wrote

Reply:
I believe so, but haven't verified it yet. Is there a good way to programmatically trigger a restart? Are unhandled exceptions sufficient or should something else be done to ensure a full restart?

Reply:
I believe unhandled exceptions are sufficient; however, I forget if O# captures those 😄 

If that doesn't work you could probably dispose our language server which may trigger an unexpected exit as well

Reply:
Unhandled exceptions / disposal just results in crashing the language server with a yellow bar. No reboot is triggered 😨. Either we have to trigger the restart another way or we may've stumbled upon a bigger issue...

Reply:
ping @NTaylorMullen

Reply:
I'll be creating a separate PR to address this (to be merged into this branch)

https://github.com/dotnet/aspnetcore/issues/30415